common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -117,18 +118,21 @@ public static String genStreamChunkId(long streamId, int chunkId) {
 
   @Override
   public void connectionTerminated(Channel channel) {
+    LinkedList<StreamState> removedStates = new LinkedList<>();
     // Close all streams which have been associated with the channel.
     for (Map.Entry<Long, StreamState> entry: streams.entrySet()) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
-        streams.remove(entry.getKey());
-
-        // Release all remaining buffers.
-        while (state.buffers.hasNext()) {
-          ManagedBuffer buffer = state.buffers.next();
-          if (buffer != null) {
-            buffer.release();
-          }
+        removedStates.add(streams.remove(entry.getKey()));
+      }
+    }
+
+    for (StreamState state: removedStates) {
+      // Release all remaining buffers.
```

Comment:
Since we're here, would we be better to try our best to clean up everything? Like adding try-catch and try to release buffers for all removedStates even there's any exception on releasing any state. 

Regarding throwing exception, we could just store the first exception if anything occurs (other exceptions could be just logged) and throw it. Actually given the implementation of `TransportRequestHandler.channelInactive`, we can even log these exceptions as ERROR and don't throw any; effectively same.

Reply:
What about log these exceptions as WARN?

Reply:
Given TransportRequestHandler.channelInactive logs the exception as ERROR I would suggest do the same, but committers can weigh it in any way while reviewing so wouldn't be a big deal for now.

Reply:
If iterator `.next()` throws an exception and we `catch` it, the iterator cursor/index won't be increased and we'll be stuck at that iterator position, and the next time that `.next()` get called it will throw the same exception.

Reply:
What I actually suggested was like:

```
for (StreamState state: removedStates) {
  try {
      while (state.buffers.hasNext()) {
        ManagedBuffer buffer = state.buffers.next();
        if (buffer != null) {
          buffer.release();
        }
      }
  } catch {
    case NonFatal(e) => logError("...proper log message with e...")
  }
}
```

instead of looping with hasNext() & next() even there's exception.

But looks like it also makes sense if and only if state.buffers.hasNext() would throw exception if state.buffers.next() throws exception. Then we can determine when to give up iterating.

Reply:
got it :man_facepalming:

Reply:
there exists an case, if the application has already finished(failed), executors' info of the app has been cleaned, but the connectionTerminated be called delay, when execute this code, it will look up the buffer with `appId` and `execId` , so the buffer cannot be got and released and still stay remain. this happend in spark 2.4.3 as I report and I also make a pr, pls help review(https://github.com/apache/spark/pull/27064), thanks